### PR TITLE
[PyTorch / Inference] Change Model server to Torchserve for PyTorch Inference for 1.6.0

### PR DESCRIPTION
**Pull request by maaquib**
Addressing comments from https://github.com/aws/deep-learning-containers/pull/426

### CODE REVIEW

File: pytorch/inference/docker/1.6.0/py3/Dockerfile.cpu [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+RUN chmod +x /usr/local/bin/dockerd-entrypoint.py
+
+ADD https://raw.githubusercontent.com/aws/aws-deep-learning-containers-utils/master/deep_learning_container.py /usr/local/bin/deep_learning_container.py
+
+RUN chmod +x /usr/local/bin/deep_learning_container.py
+
+RUN pip install --no-cache-dir "sagemaker-pytorch-inference>=2"
+
+RUN curl https://aws-dlc-licenses.s3.amazonaws.com/pytorch-1.5.1/license.txt -o /license.txt
```
Comment (YYStreet) [inline pytorch/inference/docker/1.6.0/py3/Dockerfile.cpu:96]: We can directly use the version of 1.6.0. They are already at present

File: pytorch/inference/docker/1.6.0/py3/Dockerfile.cpu [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+RUN conda install -y -c conda-forge pyyaml==5.3.1
+RUN pip install pillow==7.1.0 "awscli<2"
```
Comment (YYStreet) [inline pytorch/inference/docker/1.6.0/py3/Dockerfile.cpu:99]: Can we try to pin pillow to version 7.2.0?

File: pytorch/inference/docker/1.6.0/py3/Dockerfile.gpu [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+RUN chmod +x /usr/local/bin/dockerd-entrypoint.py
+
+ADD https://raw.githubusercontent.com/aws/aws-deep-learning-containers-utils/master/deep_learning_container.py /usr/local/bin/deep_learning_container.py
+
+RUN chmod +x /usr/local/bin/deep_learning_container.py
+
+RUN pip install --no-cache-dir "sagemaker-pytorch-inference>=2"
+
+RUN curl https://aws-dlc-licenses.s3.amazonaws.com/pytorch-1.5.1/license.txt -o /license.txt
```
Comment (YYStreet) [inline pytorch/inference/docker/1.6.0/py3/Dockerfile.gpu:111]: We can directly use the version of 1.6.0. They are already at present

File: pytorch/inference/docker/1.6.0/py3/Dockerfile.gpu [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+RUN conda install -y -c conda-forge pyyaml==5.3.1
+RUN pip install pillow==7.1.0 "awscli<2"
```
Comment (YYStreet) [inline pytorch/inference/docker/1.6.0/py3/Dockerfile.gpu:114]: Can we try to pin pillow to version 7.2.0?

File: pytorch/inference/docker/1.6.0/py3/Dockerfile.gpu [hunk 5]
```diff
@@ -0,0 +1,118 @@
+FROM nvidia/cuda:10.1-cudnn7-devel-ubuntu16.04
+# NCCL_VERSION=2.4.7, CUDNN_VERSION=7.6.2.24
+
+LABEL maintainer="Amazon AI"
+LABEL dlc_major_version="2"
```
Comment (YYStreet) [inline pytorch/inference/docker/1.6.0/py3/Dockerfile.gpu:5]: Please help us to change the "dlc_major_version" to 1

File: pytorch/inference/docker/1.6.0/py3/Dockerfile.cpu [hunk 6]
```diff
@@ -0,0 +1,103 @@
+FROM ubuntu:16.04
+
+LABEL maintainer="Amazon AI"
+LABEL dlc_major_version="2"
```
Comment (YYStreet) [inline pytorch/inference/docker/1.6.0/py3/Dockerfile.cpu:4]: Please help us to change the "dlc_major_version" to 1

**Comment by TusharKanekiDey** [comment]
CPU Inference Container is failing to collect torchvision
`

·[91m  ERROR: HTTP error 403 while getting https://torchvision-build.s3.amazonaws.com/1.6.0/cpu/torchvision-0.7.0-cp36-cp36m-manylinux1_x86_64.whl
--
949 | ERROR: Could not install requirement torchvision==0.7.0 from https://torchvision-build.s3.amazonaws.com/1.6.0/cpu/torchvision-0.7.0-cp36-cp36m-manylinux1_x86_64.whl because of HTTP error 403 Client Error: Forbidden for url: https://torchvision-build.s3.amazonaws.com/1.6.0/cpu/torchvision-0.7.0-cp36-cp36m-manylinux1_x86_64.whl for URL https://torchvision-build.s3.amazonaws.com/1.6.0/cpu/torchvision-0.7.0-cp36-cp36m-manylinux1_x86_64.whl

`

**Comment by TusharKanekiDey** [comment]
@arjkesh Can you suggest the required changes needed for the sanity test ?

**Comment by TusharKanekiDey** [comment]
The Inference Images are not able to import torch .

**Comment by tejaschumbalkar** [comment]
`Dockerfile.gpu` file should be placed in folder with name of cuda version, in this case `cu101 `. 
 
Checkout [v1.5.1](https://github.com/aws/deep-learning-containers/tree/master/pytorch/inference/docker/1.5.1/py3) for syntax

**Comment by TusharKanekiDey** [comment]
@choidongyeon has suggested that the following commit might solve the issue of torch not getting import -
https://github.com/aws/deep-learning-containers/pull/356/commits/81121552ef3a227472393c16720998491d337288
https://github.com/aws/deep-learning-containers/pull/356/commits/dce9c03fb800caf2a0f39a9514bb847c9355188f

**Comment by TusharKanekiDey** [comment]
When making Torchserve related change in test infrastructure. Please make sure that it does not affect the testing of EIA Images and Older PT Serving Images which does not have Torchserve.

File: pytorch/inference/docker/1.6.0/py3/Dockerfile.cpu [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+ && ~/miniconda.sh -b -p /opt/conda \
+ && rm ~/miniconda.sh \
+ && /opt/conda/bin/conda update conda \
+ && /opt/conda/bin/conda install -y \
+    python=$PYTHON_VERSION \
+    cython==0.29.12 \
+    ipython==7.7.0 \
+    mkl-include==2019.4 \
+    mkl==2019.4 \
+    numpy==1.16.4 \
```
Comment (TusharKanekiDey) [inline pytorch/inference/docker/1.6.0/py3/Dockerfile.cpu:62]: Can we update the numpy version? In training, we are using the latest version(1.19.1)

**Comment by YYStreet** [comment]
need to be tested on eia mode in case there is no affect on eia side

File: src/config/build_config.py [hunk 8]
```diff
@@ -1,6 +1,6 @@
 # Please only set it to True if you are preparing an EI related PR
 # Do remember to revert it back to False before merging any PR (including EI dedicated PR)
-ENABLE_EI_MODE = False
```
Comment (YYStreet) [inline src/config/build_config.py:3]: Will be reverted

Comment (TusharKanekiDey) [reply src/config/build_config.py:3]: Thanks for doing this.

**Pull request merged**